presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -78,42 +78,63 @@ public boolean matches(long numberOfRows, Map<ColumnDescriptor, Statistics<?>> s
         if (numberOfRows == 0) {
             return false;
         }
-        ImmutableMap.Builder<ColumnDescriptor, Domain> domains = ImmutableMap.builder();
+
+        if (effectivePredicate.isNone()) {
+            return false;
+        }
+
+        Map<ColumnDescriptor, Domain> effectivePredicateDomains = effectivePredicate.getDomains()
+                .orElseThrow(() -> new IllegalStateException("Effective predicate other than none should have domains"));
 
         for (RichColumnDescriptor column : columns) {
             Statistics<?> columnStatistics = statistics.get(column);
 
-            Domain domain;
             Type type = getPrestoType(effectivePredicate, column);
             if (columnStatistics == null || columnStatistics.isEmpty()) {
                 // no stats for column
-                domain = Domain.all(type);
             }
             else {
-                domain = getDomain(type, numberOfRows, columnStatistics, id, column.toString(), failOnCorruptedParquetStatistics);
+                Domain domain = getDomain(type, numberOfRows, columnStatistics, id, column.toString(), failOnCorruptedParquetStatistics);
+                if (domain.isNone()) {
+                    return false;
+                }
+                Domain effectivePredicateDomain = effectivePredicateDomains.get(column);
+                if (effectivePredicateDomain != null) {
```

Comment:
```java
if (effectivePredicateDomain != null && effectivePredicateDomain.intersect(domain).isNone()) {

Reply:
I would suggest to keep the original commit as is, and than if needed address the nits in a separate commit